os/include/tinyara/binary_manager.h

```diff
@@ -202,10 +219,16 @@ typedef struct binmgr_request_s binmgr_request_t;
 
 struct binmgr_createbin_response_s {
 	int result;
-	char binpath[CONFIG_PATH_MAX];
+	char binpath[BINARY_PATH_LEN];
 };
 typedef struct binmgr_createbin_response_s binmgr_createbin_response_t;
 
+struct binmgr_getstate_response_s {
+	int result;
```

Comment:
If you use int, not enum type, you can use half of the int for each. Because `result` and `state` cannot be over than half of the int-max.